src/fe/fe_monomial_vec.C

```diff
@@ -0,0 +1,845 @@
+
+      for (unsigned int d = 0; d < static_cast<unsigned int>(dim); d++)
+        for (unsigned int n = 0; n < n_nodes; n++)
+        {
+
+          // Zero before summation
+          nodal_soln[d + dim * n] = 0;
+
+          // u_i = Sum (alpha_i phi_i)
+          for (unsigned int i = 0; i < n_sf; i++)
+            nodal_soln[d + dim * n] += elem_soln[d + dim * i] *
+                                       FEInterface::shape(dim, fe_type, elem, i, refspace_nodes[n]);
+        }
+
+      return;
+    } // default
+  }   // switch
+}
+} // anonymous namespace
+
+// Do full-specialization for every dimension, instead
+// of explicit instantiation at the end of this file.
+// This could be macro-ified so that it fits on one line...
+template <>
+void
+FE<0, MONOMIAL_VEC>::nodal_soln(const Elem * elem,
+                                const Order order,
+                                const std::vector<Number> & elem_soln,
+                                std::vector<Number> & nodal_soln)
+{
+  FE<0, MONOMIAL>::nodal_soln(elem, order, elem_soln, nodal_soln);
+}
+
+template <>
+void
+FE<1, MONOMIAL_VEC>::nodal_soln(const Elem * elem,
+                                const Order order,
+                                const std::vector<Number> & elem_soln,
+                                std::vector<Number> & nodal_soln)
+{
+  FE<1, MONOMIAL>::nodal_soln(elem, order, elem_soln, nodal_soln);
+}
+
+template <>
+void
+FE<2, MONOMIAL_VEC>::nodal_soln(const Elem * elem,
+                                const Order order,
+                                const std::vector<Number> & elem_soln,
+                                std::vector<Number> & nodal_soln)
+{
+  monomial_vec_nodal_soln(elem, order, elem_soln, 2 /*dimension*/, nodal_soln);
+}
+
+template <>
+void
+FE<3, MONOMIAL_VEC>::nodal_soln(const Elem * elem,
+                                const Order order,
+                                const std::vector<Number> & elem_soln,
+                                std::vector<Number> & nodal_soln)
+{
+  monomial_vec_nodal_soln(elem, order, elem_soln, 3 /*dimension*/, nodal_soln);
+}
+
+// Specialize for shape function routines by leveraging scalar MONOMIAL elements
+
+// 0-D
+template <>
+RealGradient
+FE<0, MONOMIAL_VEC>::shape(const ElemType type,
+                           const Order order,
+                           const unsigned int i,
+                           const Point & p)
+{
+  Real value = FE<0, MONOMIAL>::shape(type, order, i, p);
+  return libMesh::RealGradient(value);
+}
+template <>
+RealGradient
+FE<0, MONOMIAL_VEC>::shape_deriv(const ElemType type,
+                                 const Order order,
+                                 const unsigned int i,
+                                 const unsigned int j,
+                                 const Point & p)
+{
+  Real value = FE<0, MONOMIAL>::shape_deriv(type, order, i, j, p);
+  return libMesh::RealGradient(value);
```

Comment:
Ha!  But here you're back to writing RealGradient again.  The confusion will never be eradicated.

Reply:
Haha I will change it!